src/globalsearch/spotifysearchprovider.cpp

```diff
@@ -90,30 +89,60 @@ void SpotifySearchProvider::SearchFinishedSlot(
   PendingState state = it.value();
   queries_.erase(it);
 
+  QMap<std::string, std::string> album_dedup;
```

Comment:
I think it would be better to add a comment to explain the logic we now apply here i.e. that we will check for duplicate in results.